### ISSUE TITLE
Serverless Functions: Node16 runtime broke response handling

### ISSUE THREAD

**Issue by gecko17** [kyma-project/kyma#16673]
**Description**

The Node16 broke the response handling of the serverless functions. As a result, the documented way of how to write the user function `main` is not working anymore. The user now **has to return something in the return statement**. An empty `return;` causes the server.js code to not return an http response to the caller and instead runs into a timeout. If investigating the code at https://github.com/kyma-project/kyma/blob/main/components/function-runtimes/nodejs16/server.js#L95-L133 it makes sense, why the code never terminates. 

Additionally, it is not possible anymore to send custom HTTP response codes. Obviously the code always sets hard coded response codes on the response object after the user function might already have set a response status code on the response object.

Moreover, the code is also not checking if the response was already sent by the user function with `event.extensions.response.send()`. The node14 code checked if the response was already finished by the user.

**Expected result**

Those functionalities were working in Node14 runtime:
1. Support User Function to return null
2. Support Custom HTTP Response Codes
3. Support directly sending the HTTP Response from the User Function (Because the Response Object is exposed to the User Function via the Event object, so anyone could do this and the function would crash).

**Actual result**

1. Will cause the function to not return an HTTP response. Caller will wait until timeout exceeds.
2. Not possible
3. Would break

**Steps to reproduce**
1. Use Node16 runtime and create the following inline function:
```
module.exports = {    
    main: async function (event, context) {
        console.log("Triggered");
        return;
    }
}
```
2. Call the function via the created service with an HTTP Post request, request will not terminate.
3. Change function to set the response code:
```
module.exports = {    
    main: async function (event, context) {
        console.log("Triggered");
        event.extensions.response.status(404);
        return "not found";
    }
}
3. Call again and observe that the response will not have status code 404.
```

**Troubleshooting**

There code for calling the user function has a lot of bugs and is not handling all cases correctly:
https://github.com/kyma-project/kyma/blob/main/components/function-runtimes/nodejs16/server.js#L95-L133

Have a look at lines 113-115 especially: If the result is `null`, which it will be when not returning anything in the user function, the callback will not be called and no HTTP response will be sent.

**Comment by kwiatekus** [comment]
Related: https://github.com/kyma-project/kyma/issues/16462

**Comment by gecko17** [comment]
The fix for #16462 is not working for async functions, that return a promise.
